design/general_refactorings.md

```diff
@@ -0,0 +1,49 @@
+Before refactor concrete thing here is a list of things that need to be done all over the code:
+
+1. Go Over All Namings And Improve them
+
+2. Directory Organize:
+
+   Current Directories
+
+   | Directory Name | Porpuse                           |
+   | :------------- | :-------------------------------- |
+   | host           | files related to host command     |
+   | init           | files related to init command     |
+   | list           | files related to list command     |
+   | login          | files related to login command    |
+   | logout         | files related to logout command   |
+   | policy         | files related to policy command   |
+   | user           | files related to user command     |
+   | variable       | files related to variable command |
+
+   
+
+   And this files are without a folder : api.py, client.py, argparse_wrapper.py, cli.py , client.py, config.py, constants.py, credentials_data.py, credentials_form_file.py, endpoint.py, errors.py, http_wrapper.py, ssl_service.py, version.py
+
+   The change i want to make is to organize the files by their types and not by the command they do and have directories that can be used for the files without directory:
+
+   
+
+   | Directory Name | Porpuse                                                      |
+   | :------------- | :----------------------------------------------------------- |
+   | utils          | Wrappers and utils of diffrent python libsssl_servicearg_parse_wrapperhttp_wrapper |
+   | api            | Files that do the directly talk with conjurclient.pyapi.py   |
+   | logics         | All logic classes here                                       |
+   | controllers    | all controller classes                                       |
+   | logics         | all logic classes                                            |
+   | data_objects   | all data objects classes                                     |
+
+   
+
+3. Go over thrown exceptions in the code and change them to our own exceptions
```

Comment:
Error handling

Again, follow the template for readability

## Error handling

_Proposal_
Map current error and exception handling and create Conjur-specific exceptions

_Current state_
Through the codebase, we have very general errors that are not representative of the actual error that took place... <fill in more info>

_Recommendation_
Map out the current error flow and convert general exceptions to Conjur-specific exceptions. This will <explain why this is a great change>

Reply:
Please provide examples in code. Maybe add Example section under _Current state_